src/CTA.FeatureDetection.Common/WCFConfigUtils/CoreWCFParityCheck.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using CTA.FeatureDetection.Common.Models.WCF;
+
+namespace CTA.FeatureDetection.Common.WCFConfigUtils
+{
+    public class CoreWCFParityCheck
+    {
+        /// <summary>
+        /// Check if given Transport and Mode Collection has Parity with CoreWCF.
+        /// If one binding and transport has support in CoreWCF, the method returns true.
+        /// </summary>
+        /// <param name="bindingsTransportMap">Dictionary with Binding Name as key and list of modes as value</param>
+        /// <returns>If the given bindings and transport mode has WCF parity.</returns>
+        public static bool HasCoreWCFParity(Dictionary<string, BindingConfiguration> bindingsTransportMap)
+        {
+            bool hasCoreWCFSupport = false;
+
+            foreach (var binding in bindingsTransportMap)
+            {
+                var bindingName = binding.Key;
+
+                //Variables assigned but not used, can be used as a metric
+                var unsupportedBindings = new List<string>();
+                var unsupportedModes = new Dictionary<string, string>();
+
+                if (CoreWCFBindings.CORE_WCF_BINDINGS.Keys.Contains(bindingName))
+                {
+                    var mode = bindingsTransportMap[bindingName].Mode;
+                    var supportedModes = CoreWCFBindings.CORE_WCF_BINDINGS[bindingName];
+
+                    if (!supportedModes.Contains(mode.ToLower()))
```

Comment:
Is it always lower case? Do we need to do a case insensitive comparison?

Reply:
+1

Reply:
The supportedModes is a constants which I have defined in lower case, because various configurations may have different type casing for binding name and mode, I have kept case insensitive comparison